.github/workflows/build-interop-docker.yml

```diff
@@ -35,10 +36,20 @@ jobs:
             echo 'gitref=${{ github.sha }}' | tee -a $GITHUB_OUTPUT;
           fi
       - uses: docker/build-push-action@v4
+        id: docker_build_and_push
         with:
           context: "{{defaultContext}}:interop"
           platforms: linux/amd64,linux/arm64
-          push: true
+          push: ${{ github.event_name != 'pull_request' }}
           build-args: |
             GITREF=${{ steps.tag.outputs.gitref }}
           tags: martenseemann/quic-go-interop:${{ steps.tag.outputs.tag }}
+      - name: Run QUIC Interop tests
```

Comment:
You might want to make this a separate step:
1. Building the image is cheap (this is Go after all ;))
2. The run triggers are orthogonal anyway, so there's no condition where we'd build twice anyway.

Reply:
> You might want to make this a separate step:

Note that it is already a separate step. Are you suggesting to make it a separate job instead?

Reply:
Yes, separate job is what I meant, sorry for mixing up the terminology here. I think I'd even prefer a separate workflow, since it's (almost) orthogonal to this one.